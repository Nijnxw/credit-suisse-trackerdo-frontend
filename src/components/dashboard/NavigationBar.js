import { Alignment, Classes, Icon, Navbar, NavbarDivider, NavbarGroup, NavbarHeading } from '@blueprintjs/core'
import { IconNames } from '@blueprintjs/icons'
import classNames from 'classnames'
import * as React from 'react'
import { NavLink } from 'react-router-dom'
import ProgressMenu from '../progress/components/ProgressMenu'
import AddIcon from '@material-ui/icons/Add';
import NotificationsIcon from '@material-ui/icons/Notifications';

const NavigationBar = () => {
  return (
    <Navbar className={classNames("NavigationBar primary-navbar pt-dark")}>
      <NavbarGroup align={Alignment.LEFT}>
        <NavLink
          activeClassName="pt-active"
          className={classNames("NavigationBar__link pt-button pt-minimal", Classes.BUTTON)}
          to="/"
        >
          <Icon icon={IconNames.BOOK} />
          <NavbarHeading className="hidden-xs">Trackerdo</NavbarHeading>
        </NavLink>
      </NavbarGroup>

      <NavbarGroup align={Alignment.RIGHT}>
        {/* PROGRESS MENU */}
        <ProgressMenu />
        {/* <NavLink
          activeClassName="pt-active"
          className="NavigationBar__link pt-button pt-minimal"
          to="/app/progress"
        >
          <Icon icon={IconNames.TRENDING_UP} />
        </NavLink> */}

        <div className="hidden-xs">
          <NavbarDivider className="default-divider" />
        </div>

        {/* ADD TASK */}
        <NavLink
          activeClassName="pt-active"
          className="NavigationBar__link pt-button pt-minimal"
          to="/app/add-task"
        >
          <AddIcon fontSize='small' />
        </NavLink>
        
        <div className="hidden-xs">
          <NavbarDivider className="default-divider" />
        </div>

        {/* NOTIFICATION BUTTON */}
        <NavLink
          activeClassName="pt-active"
          className="NavigationBar__link pt-button pt-minimal"
          to="/app/notifications"
        >
            <NotificationsIcon fontSize='small' />
        </NavLink>

        <div className="hidden-xs">
          <NavbarDivider className="default-divider" />
        </div>
        
        {/* TODO: profiling menu here after authentication */}

      </NavbarGroup>
    </Navbar>
  )
}

export default NavigationBar;
