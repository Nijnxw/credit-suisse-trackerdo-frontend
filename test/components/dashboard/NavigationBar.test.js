import React from 'react'
import { shallow } from 'enzyme'
import NavigationBar from '../../../src/components/dashboard/NavigationBar'
import { Navbar, NavbarGroup } from '@blueprintjs/core'
import { NavLink } from 'react-router-dom'

let wrapper

beforeEach(() => {
  wrapper = shallow(<NavigationBar /> );
})

describe('<NavigationBar /> rendering', () => {
  it('should render correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  })
  
  it('should render 1 <NavBar>', () => {
    expect(wrapper.find(Navbar)).toHaveLength(1);
  })

  it('should render 2 <NavBarGroup>', () => {
    expect(wrapper.find(NavbarGroup)).toHaveLength(2);
  })

  it('should render 3 <NavLink>', () => {
    expect(wrapper.find(NavLink)).toHaveLength(3);
  })
})
