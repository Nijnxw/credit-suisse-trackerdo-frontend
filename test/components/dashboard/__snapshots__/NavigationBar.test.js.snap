// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> rendering should render correctly 1`] = `
"<Blueprint3.Navbar className=\\"NavigationBar primary-navbar pt-dark\\">
  <Blueprint3.NavbarGroup align=\\"left\\">
    <NavLink activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal bp3-button\\" to=\\"/\\">
      <Blueprint3.Icon icon=\\"book\\" />
      <Blueprint3.NavbarHeading className=\\"hidden-xs\\">
        Trackerdo
      </Blueprint3.NavbarHeading>
    </NavLink>
  </Blueprint3.NavbarGroup>
  <Blueprint3.NavbarGroup align=\\"right\\">
    <ProgressMenu />
    <div className=\\"hidden-xs\\">
      <Blueprint3.NavbarDivider className=\\"default-divider\\" />
    </div>
    <NavLink activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" to=\\"/app/add-task\\">
      <Memo(ForwardRef(AddIcon)) fontSize=\\"small\\" />
    </NavLink>
    <div className=\\"hidden-xs\\">
      <Blueprint3.NavbarDivider className=\\"default-divider\\" />
    </div>
    <NavLink activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" to=\\"/app/notifications\\">
      <Memo(ForwardRef(NotificationsIcon)) fontSize=\\"small\\" />
    </NavLink>
    <div className=\\"hidden-xs\\">
      <Blueprint3.NavbarDivider className=\\"default-divider\\" />
    </div>
  </Blueprint3.NavbarGroup>
</Blueprint3.Navbar>"
`;
